parsec/core/cli/invitation.py

```diff
@@ -0,0 +1,395 @@
+            initial_ctx = DeviceGreetInitialCtx(cmds=cmds, token=token)
+            do_greet = partial(_do_greet_device, device, initial_ctx)
+
+        while True:
+            try:
+                greet_done = await do_greet()
+                if greet_done:
+                    break
+            except InviteError as exc:
+                click.secho(str(exc), fg="red")
+            click.secho("Restarting the invitation process", fg="red")
+
+
+@click.command(short_help="greet invitation")
+@core_config_and_device_options
+@click.argument("token", type=UUID)
+def greet_invitation(config, device, token, **kwargs):
+    """
+    Greet a new device or user into the organization
+    """
+    with cli_exception_handler(config.debug):
+        # Disable task monitoring given user prompt will block the coroutine
+        trio_run(_greet_invitation, config, device, token, monitor_tasks=False)
+
+
+async def _do_claim_user(initial_ctx):
+    async with spinner("Initializing connection with greeter for claiming user"):
+        in_progress_ctx = await initial_ctx.do_wait_peer()
+
+    choices = in_progress_ctx.generate_greeter_sas_choices(size=3)
+    for i, choice in enumerate(choices):
+        display_choice = click.style(str(choice), fg="yellow")
+        click.echo(f" {i} - {display_choice}")
+    code = click.prompt(
+        f"Select code provided by greeter", type=click.Choice([str(x) for x in range(len(choices))])
+    )
+    if choices[int(code)] != in_progress_ctx.greeter_sas:
+        click.secho("Wrong code provided", fg="red")
+        return None
+
+    in_progress_ctx = await in_progress_ctx.do_signify_trust()
+    display_claimer_sas = click.style(str(in_progress_ctx.claimer_sas), fg="yellow")
+    click.echo(f"Code to provide to greeter: {display_claimer_sas}")
+    async with spinner("Waiting for greeter"):
+        in_progress_ctx = await in_progress_ctx.do_wait_peer_trust()
+
+    requested_label = click.prompt("User fullname")
+    requested_email = initial_ctx.claimer_email
+    default_device_id = f"{requested_email.split('@', 1)[0]}@{platform.node()}"
+    requested_device_id = click.prompt("Device ID", default=default_device_id, type=DeviceID)
+    async with spinner("Waiting for greeter (finalizing)"):
+        new_device = await in_progress_ctx.do_claim_user(
+            requested_device_id=requested_device_id,
+            requested_human_handle=HumanHandle(email=requested_email, label=requested_label),
+        )
+
+    return new_device
+
+
+async def _do_claim_device(initial_ctx):
+    async with spinner("Initializing connection with greeter for claiming device"):
+        in_progress_ctx = await initial_ctx.do_wait_peer()
+
+    choices = in_progress_ctx.generate_greeter_sas_choices(size=3)
+    for i, choice in enumerate(choices):
+        display_choice = click.style(str(choice), fg="yellow")
+        click.echo(f" {i} - {display_choice}")
+    code = click.prompt(
```

Comment:
Is the sync prompt gonna be a problem at some point? Maybe wrap it with `trio.to_thread.run_sync()`?

Too bad `prompt-toolkit` doesn't support trio :)

Reply:
The cli commands that make use of user input are run with task monitor disabled (`trio_run(_greet_invitation, config, device, token, monitor_tasks=False)`) in order to avoid meaningless warnings.

I don't remember clearly the issue, but using `trio.to_thread.run_sync()` to run the prompt wasn't possible. I'll retry this to investigate further ;-)

Reply:
> I don't remember clearly the issue, but using trio.to_thread.run_sync() to run the prompt wasn't possible. I'll retry this to investigate further ;-)

Oh really? I'm quite curious about it!

Reply:
No troubles in fact, I've change the code ;-)